src/libraries/System.Private.CoreLib/src/System/Threading/ReaderWriterLockSlim.cs

```diff
@@ -515,7 +515,7 @@ private bool TryEnterWriteLockCore(TimeoutTracker timeout)
                 }
             }
 
-            bool retVal = true;
+            bool retVal;
```

Comment:
Rather than defining this all the way up here, each of the two call sites later on that use this could be changed, e.g. from
```C#
                    retVal = WaitOnEvent(_waitUpgradeEvent, ref _numWriteUpgradeWaiters, timeout, EnterLockType.UpgradeToWrite);
                    // The lock is not held in case of failure.
                    if (!retVal)
                        return false;
```
to
```C#
                    bool retVal = WaitOnEvent(_waitUpgradeEvent, ref _numWriteUpgradeWaiters, timeout, EnterLockType.UpgradeToWrite);
                    // The lock is not held in case of failure.
                    if (!retVal)
                        return false;
```
and then the local could be further removed entirely:
```C#
                    // The lock is not held in case of failure.
                    if (!WaitOnEvent(_waitUpgradeEvent, ref _numWriteUpgradeWaiters, timeout, EnterLockType.UpgradeToWrite))
                        return false;
```